spec/dynamoid/persistence_spec.rb

```diff
@@ -313,4 +313,90 @@
       }
     end
   end
+
+  describe 'class-type fields' do
+    subject { doc_class.new }
+
+    context 'when Money can load itself and Money instances can dump themselves with Dynamoid-specific methods' do
+      let(:doc_class) do
+        Class.new do
+          def self.name; 'Doc'; end
+
+          include Dynamoid::Document
+
+          field :price, MoneyInstanceDump
+        end
+      end
+
+      before(:each) do
+        subject.price = MoneyInstanceDump.new(BigDecimal.new('5'))
+        subject.save!
+      end
+
+      it 'round-trips using Dynamoid-specific methods' do
+        expect(doc_class.all.first).to eq subject
+      end
+
+      it 'is findable as a string' do
+        expect(doc_class.where(price: '5.0').first).to eq subject
+      end
+    end
+
+    context 'when MoneyAdapter dumps/loads a class that does not directly support Dynamoid\'s interface' do
+      let(:doc_class) do
+        Class.new do
+          def self.name; 'Doc'; end
+
+          include Dynamoid::Document
+
+          field :price, MoneyAdapter
+        end
+      end
+
+      before(:each) do
+        subject.price = Money.new(BigDecimal.new('5'))
+        subject.save!
+        subject.reload
+      end
+
+      it 'round-trips using Dynamoid-specific methods' do
```

Comment:
There are enough differences that I don't see a good way of factoring out the commonality. These three scenarios share only one spec: "it 'round-trips using Dynamoid-specific methods'"... I don't want to make a shared example block for just one spec.